### ISSUE TITLE
Calculate VAT on entry of cashbook payments and purchase invoice lines

### ISSUE THREAD

**Issue by steveblamey** [uzerpllp/uzerp#119]
Discussion from Slack:

mshiner [12:17 PM] 
When entering a PL Invoice it might be useful if the system calculated the VAT and showed it to the user - the dialog box will need an ajax call to get the vat rate based on the drop down - we may need to move this so that it recalculates as an on change on either the net or VAT . Actually... if you look at the expenses line dialog it does something similar with Gross/Net VAT

We are talking about VAT calculation on Cashbook Payments, but I see it's the same issue with the PL Invoice. I guess that auto calc VAT needs to be in both places?

mshiner [2:50 PM] 
Well... yes and no... as I said, you are only supposed to input the VAT you are given so I'd decided to not make the calculation. If we are going to do it we need to go both from net->gross and gross>net so ajaxing both. The problem is.... what if the calc is a penny out (or more if the supplier makes an error the VAT calc) - the ajax will override your input won't it? Consider the following

Net = £1.00 supplier invoices £1.19 - you enter £1 net it calculates £1.20 you then override to make the VAT £0.19 making the gross £1.19 and the ajax does £1.19/1.2 making net £0.99 and so on. You get my drift?

steveblamey [2:56 PM] 
I see. Some extra ui logic should sort that, e.g. if you type a figure in Tax Value the tax rate selection is disabled and any selected rate is ignored. If you don't enter a Tax Value, or remove it, the tax value is calculated using the selected rate. Something like that.

----- May 6th -----
mshiner [10:12 AM] 
@steveblamey: another thing came to mind (I put this on the ticket)
"The other issue is that some suppliers (especially overseas) will not b charging VAT. If a user imports a lot then they don't really want VAT calculated."

[4:02] 
@steveblamey: on the VAT thing... I can see that its of benefit... in the SL the VAT is automatically calculated based on rate (the product) and customer. I guess to get around the VAT being calculated on PIs we'd need to take the same approach... i.e. take a look at the tax status of the supplier and decide if we need VAT (even if they leave it as std - which they need to do to get it on the VAT report)

**Comment by steveblamey** [comment]
Agreed that this is a good thing for Purchase Invoices, but not for Cashbook Payments

**Comment by mshiner** [comment]
Steve

I just found a 'bug' in this area..... When you enter a PI line you can put in VAT even if the tax status of the supplier is EU (or overseas import for that matter). This causes errors if the supplier is EU becuasse the apply EU tax rules mean you get extra postings in the VAT Acquisition control account as well as the VAT input tax control account.

I think we should pop up a warning if the user enters a value into the VAT box if the tax status of the supplier is EU or Import (i.e. the Apply Tax flag is set to false on the tax status). I wouldn't want to actually forbid the entry as it may be correct in some jurisdictions

This will also apply for reverse charge when we implement I think.
